src/documents/RemoteModelAPI.md

```diff
@@ -0,0 +1,364 @@
+---
+title: RemoteModel API
+category: Infusion
+---
+
+`fluid.remoteModelComponent` builds on top of [`fluid.modelComponent`](ComponentGrades.md) with the purpose of providing
+a buffer between a local and remote model that are attempting to stay in sync. For example, a local model is being
+updated by user interaction, this is sent back to a remote server, which in turn attempts to update the local model. If
+additional user actions occur during this roundtrip, an infinite loop of updates may occur. `fluid.remoteModelComponent`
+solves this by restricting reading and writing to a single request at a time, waiting for one request to complete before
+operating the next. Additionally, it will rebase local and remote changes, with the local changes taking priority. (See:
+[Fetch Workflow](#fetch-workflow) for more details on the rebasing.)
+
+## Supported Events
+
+In addition to the standard `fluid.modelComponent` events, `fluid.remoteModelComponent` adds the following:
+
+* `onFetch`,
+* `onFetchError`,
+* `afterFetch`,
+* `onWrite`,
+* `onWriteError`,
+* `afterWrite`
+
+For information on how events work and how to configure listeners to them, see [Infusion Event System](InfusionEventSystem.md).
+
+<div class="infusion-docs-note">
+    <strong>Note:</strong> The <code>onFetch</code>, <code>afterFetch</code>, <code>onWrite</code> and
+    <code>afterWrite</code> are <a href="PromisesAPI.md#fluidpromisefiretransformeventevent-payload-options">synthetic
+    events</a> that fire as a sequence. As implemented in the <code>fluid.remoteModelComponent</code>, the listeners for
+    the event will all fire in the specified order and be passed the same arguments. However, the next listener will not
+    trigger till the previous one has completed. That is, either returned a value or resolved/rejected a returned
+    promise. This is useful to prevent further actions from happening until all of the listeners have completed, but
+    unlike other <a href="PromisesAPI.md#fluidpromisefiretransformeventevent-payload-options">event sequences</a>, it
+    cannot be used to pass the payload through a series of transformations.
+</div>
+
+### onFetch
+
+<table>
+    <tbody>
+        <tr>
+            <th>Description</th>
+            <td>
+                Fires an event sequence that executes before <code>fetchImpl</code> is called.
+            </td>
+        </tr>
+        <tr>
+            <th>Type</th>
+            <td>default</td>
+        </tr>
+        <tr>
+            <th>Parameters</th>
```

Comment:
"Parameters" in all these blocks should really be "arguments" since we are trying to document what the caller should provide